packages/orm/src/relations/hasPersistedAsyncRelation.ts

```diff
@@ -0,0 +1,342 @@
+    setField(this.#entity, this.fieldName, isEntity(other) ? other?.idTagged ?? other : other);
+
+    if (typeof other === "string") {
+      this.loaded = undefined;
+      this._isLoaded = false;
+    } else {
+      this.loaded = other;
+      this._isLoaded = true;
+    }
+    this.maybeRemove(previous);
+    this.maybeAdd();
+  }
+
+  /** Returns the tagged id of the current value. */
+  private get idTagged(): IdOf<U> | N {
+    ensureNotDeleted(this.#entity, "pending");
+    return maybeResolveReferenceToId(this.current()) as IdOf<U> | N;
+  }
+
+  /** Returns the id of the current value. */
+  get id(): IdOf<U> | N {
+    ensureNotDeleted(this.#entity, "pending");
+    // If current is a string, we might need to detag it...
+    const id = maybeResolveReferenceToId(this.current()) as IdOf<U> | N;
+    if (!this.otherMeta.idTagged && id) {
+      return deTagId(this.otherMeta, id) as IdOf<U>;
+    }
+    return id;
+  }
+
+  get idOrFail(): IdOf<U> {
+    ensureNotDeleted(this.#entity, "pending");
+    return (this.id as IdOf<U> | undefined) || fail("Reference is unset or assigned to a new entity");
+  }
+
+  get idUntagged(): string | undefined {
+    return this.id && deTagIds(this.otherMeta, [this.id])[0];
+  }
+
+  get idUntaggedOrFail(): string {
+    return this.idUntagged || fail("Reference is unset or assigned to a new entity");
+  }
+
+  get loadHint(): any {
+    return getMetadata(this.#entity).config.__data.cachedReactiveLoadHints[this.fieldName];
+  }
+
+  // private impl
+
+  setFromOpts(other: U | IdOf<U> | N): void {
+    this.setImpl(other);
+  }
+
+  initializeForNewEntity(): void {
+    // We can be initialized with [entity | id | undefined], and if it's entity or id, then setImpl
+    // will set loaded appropriately; but if we're initialized undefined, then mark loaded here
+    if (this.current() === undefined) {
+      this._isLoaded = true;
+    }
+  }
+
+  maybeCascadeDelete(): void {
+    if (this.isCascadeDelete) {
+      const current = this.current({ withDeleted: true });
+      if (current !== undefined && typeof current !== "string") {
+        this.#entity.em.delete(current as U);
+      }
+    }
+  }
+
+  async cleanupOnEntityDeleted(): Promise<void> {
+    const current = await this.load({ withDeleted: true });
+    if (current !== undefined) {
+      const o2m = this.getOtherRelation(current);
+      if (o2m instanceof OneToManyCollection) {
+        o2m.remove(this.#entity, { requireLoaded: false });
+      } else if (o2m instanceof OneToManyLargeCollection) {
+        o2m.remove(this.#entity);
+      } else if (o2m instanceof OneToOneReferenceImpl) {
+        o2m.set(undefined as any);
+      } else {
+        throw new Error(`Unhandled ${o2m}`);
+      }
+    }
+    setField(this.#entity, this.fieldName, undefined);
+    this.loaded = undefined as any;
+    this._isLoaded = true;
+  }
+
+  maybeRemove(other: U | undefined) {
+    if (other) {
```

Comment:
Thought: Maybe these go away? ... or maybe we just don't generate `Book.favoritedAuthors` for now until we can put more time into making it super amazing. Or maybe we never need it.